packages/openapi3/src/schema-emitter.ts

```diff
@@ -645,6 +651,63 @@ export class OpenAPI3SchemaEmitter extends TypeEmitter<
     }
   }
 
+  #attachXmlObject(program: Program, prop: ModelProperty, emitObject: OpenAPI3Schema) {
+    const xmlObject: OpenAPI3XmlSchema = {};
+
+    const isXmlModel =
+      prop.model && prop.model?.name !== resolveEncodedName(program, prop.model, "application/xml");
+
+    // Resolve XML name
+    const xmlName = resolveEncodedName(program, prop, "application/xml");
+    if (xmlName !== prop.name) {
+      xmlObject.name = xmlName;
+    }
+
+    // Get and set XML namespace if present
+    const currNs = getNs(program, prop);
+    if (currNs) {
+      xmlObject.prefix = currNs.prefix;
+      xmlObject.namespace = currNs.namespace;
+    }
+
+    // Set XML attribute if present
+    if (isAttribute(program, prop)) {
+      xmlObject.attribute = true;
+    }
+
+    // Handle array wrapping if necessary
+    if (
+      prop.type?.kind === "Model" &&
+      isArrayModelType(program, prop.type) &&
+      (xmlObject.name || isXmlModel)
+    ) {
+      xmlObject.wrapped = true;
+    }
+
+    // Set XML unwrapped if present
+    if (isUnwrapped(program, prop)) {
+      xmlObject.wrapped = false;
+    }
+
+    // if wrapped is false, xml.name of the wrapping element is ignored.
+    if (xmlObject.wrapped === false) {
+      xmlObject.name = undefined;
+    }
+
+    // only apply properties with the value is true.
+    const schema = new ObjectBuilder<OpenAPI3XmlSchema>();
```

Comment:
i don't think this is necssary you can just set xmlObject line 707

Reply:
try not to set something like `prop: undefined`

Reply:
Should we check and limit the decorators to the expected objects? if yes, maybe add the limit in another pr?